From bed0542b2f6ae371596895c4487c33c345f85897 Mon Sep 17 00:00:00 2001
From: Sagar Ghuge <sagar.ghuge@intel.com>
Date: Wed, 11 Oct 2023 13:59:21 -0700
Subject: [PATCH 202/834] anv: Enable transfer queue only on ACM+ platforms

On older platforms, we have the blitter engine, but it lacks compression
handling and other features we need, unfortunately, so enable the
transfer queue only on ACM+ platforms.

Signed-off-by: Sagar Ghuge <sagar.ghuge@intel.com>
Reviewed-by: Lionel Landwerlin <lionel.g.landwerlin@intel.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25667>
---
 src/intel/vulkan/anv_device.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/intel/vulkan/anv_device.c b/src/intel/vulkan/anv_device.c
index 93092047f62..6713e778f86 100644
--- a/src/intel/vulkan/anv_device.c
+++ b/src/intel/vulkan/anv_device.c
@@ -1152,7 +1152,8 @@ anv_physical_device_init_queue_families(struct anv_physical_device *pdevice)
          c_count < 1 ? INTEL_ENGINE_CLASS_RENDER : INTEL_ENGINE_CLASS_COMPUTE;
 
       int blit_count = 0;
-      if (debug_get_bool_option("INTEL_COPY_CLASS", false)) {
+      if (debug_get_bool_option("INTEL_COPY_CLASS", false) &&
+          pdevice->info.verx10 >= 125) {
          blit_count = intel_engines_count(pdevice->engine_info,
                                           INTEL_ENGINE_CLASS_COPY);
       }
-- 
2.42.0

